web/src/main/java/com/epam/rft/atsy/web/controllers/ApplicationStateController.java

```diff
@@ -77,16 +90,25 @@ public ModelAndView loadPage(@RequestParam Long applicationId,
             .convert(statesHistoryService.getStateHistoriesByApplicationId(applicationId),
                 StateHistoryViewRepresentation.class);
 
+
     if (clickedState != null) {
+      ApplicationDTO applicationDTO = applicationsService.getApplicationDtoById(applicationId);
       StateDTO clickedStateDTO = stateService.getStateDtoByName(clickedState);
 
       Assert.notNull(clickedStateDTO);
 
-      stateHistoryViewRepresentations.add(0, StateHistoryViewRepresentation.builder()
+      StateHistoryViewRepresentation representation = StateHistoryViewRepresentation.builder()
           .creationDate(new Date())
           .stateId(clickedStateDTO.getId())
           .stateName(clickedStateDTO.getName())
-          .build());
+          .build();
+
+      if (clickedStateDTO.getId() == 1) {
```

Comment:
Write comment about this line, because it seems some kind of magic.

You and I understand what this line means now, but it can be confusing for others or it can be confusing for us later. Another opportunity: use StateDTO's name field if it is unique, it can be more readable.